File: frontend/src/api/ai.ts
import { api } from "@/lib/api";

export type AIAction =
  | "improve_writing"
  | "generate_caption"
  | "rewrite"
  | "translate"
  | "generate_hashtags"
  | "change_tone";

export interface AIAssistResult {
  text: string;
  is_mock: boolean;
}

export async function runAIAction(action: AIAction, text: string, options: Record<string, string> = {}) {
  const { data } = await api.post<AIAssistResult>("/posts/ai-assist", { action, text, options });
  return data;
}

File: backend/app/schemas/ai.py
from typing import Literal

from pydantic import BaseModel

AIAction = Literal[
    "improve_writing", "generate_caption", "rewrite", "translate", "generate_hashtags", "change_tone"
]


class AIAssistRequest(BaseModel):
    action: AIAction
    text: str = ""
    options: dict = {}


class AIAssistResponse(BaseModel):
    text: str
    is_mock: bool
